### PR TITLE
V3 customer case UI - top section

### PR DESCRIPTION
**Pull request by anemne**
This PR includes updates to the top section of customer cases, aligned with the finalized design. 

Screenshot 2024-10-25 at 14 41 20

### CODE REVIEW

File: src/components/customerCases/customerCase/customerCase.module.css [hunk 1]
```diff
@@ -8,33 +8,75 @@
 .content {
   display: flex;
   flex-direction: column;
-  max-width: 1200px;
```
Comment (mathiazom) [inline src/components/customerCases/customerCase/customerCase.module.css:11]: Why remove the max-width? Makes it pretty hard to read

image

File: src/components/text/text.module.css [hunk 2]
```diff
@@ -60,7 +60,9 @@
   }
 }
 
-/* TODO: we need to add sizes og h4-h6 */
+/* .h4 */
+/* .h5 */
+/* .h6 */
```
Comment (mathiazom) [inline src/components/text/text.module.css:65]: Should these be defined later? Some TODO comments would be nice

File: studioShared/schemas/documents/customerCase.ts [hunk 3]
```diff
@@ -32,7 +32,29 @@ const customerCase = defineType({
       type: "internationalizedArrayText",
       title: "Description",
       description:
-        "Short paragraph displayed at the top of the customer case page",
+        "Short paragraph displayed at the top of the customer case page. Description can not be more than 50 words.",
```
Comment (mathiazom) [inline studioShared/schemas/documents/customerCase.ts:35]: Why word count instead of characters?

File: studioShared/lib/queries/customerCases.ts [hunk 4]
```diff
@@ -36,7 +36,9 @@ export const CUSTOMER_CASE_QUERY = groq`
       "name": ${translatedFieldFragment("name")},
       "duration": ${translatedFieldFragment("duration")},
       "sector": ${translatedFieldFragment("sector")},
-      "delivery": ${translatedFieldFragment("delivery")},
+      "deliveries": delivery[] {
+        "delivery": ${translatedFieldFragment("delivery")},
+      },
```
Comment (mathiazom) [inline studioShared/lib/queries/customerCases.ts:41]: We should be consistent with "deliveries"/"delivery" in schema and GROQ response. I vote for "deliveries" everywhere for the array, and "delivery" for the string value of each array item.

File: src/styles/global.css [hunk 5]
```diff
@@ -40,9 +40,13 @@ html {
 body {
   margin: 0;
   padding: 0;
+  overflow-x: hidden;
```
Comment (mathiazom) [inline src/styles/global.css:43]: Why is this needed? If there is accidental overflowing somewhere, we should fix it there instead.

File: studioShared/schemas/fields/customerCaseProjectInfo.ts [hunk 6]
```diff
@@ -31,7 +34,36 @@ export const customerCaseProjectInfo = defineField({
...
+              name: "delivery",
+              title: "Delivery",
+              type: "internationalizedArrayString",
+            },
+          ],
+          preview: {
+            select: {
+              item: "delivery",
+            },
+            prepare({ item }) {
```
Comment (mathiazom) [inline studioShared/schemas/fields/customerCaseProjectInfo.ts:54]: `item` is very generic, how about just naming it `delivery`?

File: src/components/customerCases/customerCase/customerCase.module.css [hunk 7]
```diff
@@ -8,33 +8,75 @@
...
+
+  @media (max-width: 1024px) {
+    max-width: 100%;
+  }
 }
 
 .projectInfo {
   display: flex;
+  justify-content: space-between;
+  align-items: center;
```
Comment (mathiazom) [inline src/components/customerCases/customerCase/customerCase.module.css:58]: Project info should not be center-aligned (only has effect on mobile)

image

File: src/components/customerCases/customerCases.module.css [hunk 8]
```diff
@@ -9,8 +9,6 @@
   display: flex;
   flex-direction: column;
   gap: 5rem;
-  max-width: 1200px;
```
[the remark this answers is not in the document]
Comment (anemne) [reply src/components/customerCases/customerCases.module.css:12]: If we set a max-width on this section it will become much more narrow than what's in the sketches. I am, however, using a wide screen here and we might want to set a max-width. I'll ask Julia or OP. 

with max width: 
Screenshot 2024-10-28 at 10 38 14

In sketches: 
Screenshot 2024-10-28 at 10 39 17

File: src/styles/global.css (hunk 5 above)
Comment (anemne) [reply src/styles/global.css:43]: You right! I think i forgot to remove this while testing some stuff. I'll fix it now.

File: src/components/customerCases/customerCases.module.css (hunk 8 above)
[the remark this answers is not in the document]
Comment (anemne) [reply src/components/customerCases/customerCases.module.css:12]: Max-width will be 1400px

File: src/components/customerCases/customerCase/customerCase.module.css
Comment (mathiazom) [inline src/components/customerCases/customerCase/customerCase.module.css:1]: Since the header is changed, we should reduce the top margin for `.wrapper` (currently at `8rem`), e.g. `4rem`. Could also decrease it further on mobile (e.g. `2rem`).

File: src/styles/global.css [hunk 9]
```diff
@@ -43,6 +43,9 @@ body {
   width: 100vw;
   background-color: var(--primary-bg);
 }
+main {
+  padding: 1rem;
```
Comment (mathiazom) [inline src/styles/global.css:47]: Do we really need this at the global level? This limits what we can do further down (e.g. display image at full page width).

File: studioShared/schemas/documents/customerCase.ts [hunk 10]
```diff
@@ -32,7 +32,29 @@ const customerCase = defineType({
...
+        "Short paragraph displayed at the top of the customer case page. Description can not be more than 300 characters.",
+      validation: (rule) =>
+        rule.custom<{ value: string; _type: string; _key: string }[]>(
+          (value) => {
+            if (!value) return true;
+            if (value) {
+              const invalidItems = value.filter(
+                (item) =>
+                  typeof item.value === "string" &&
+                  item.value.length - 1 >= 300,
```
Comment (mathiazom) [inline studioShared/schemas/documents/customerCase.ts:44]:
```suggestion
                  item.value.length > 300,
```

File: studioShared/schemas/documents/customerCase.ts [hunk 11]
```diff
@@ -32,7 +32,29 @@ const customerCase = defineType({
...
+              );
+
+              if (invalidItems.length) {
```
Comment (mathiazom) [inline studioShared/schemas/documents/customerCase.ts:47]: Same result, but more readable imo (less js "falsy" magic)
```suggestion
              if (invalidItems.length > 0) {
```

**Review by mathiazom — APPROVED** [review]
😎
